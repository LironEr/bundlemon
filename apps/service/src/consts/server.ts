export const MAX_BODY_SIZE_BYTES = 1024 * 1024; // 1MB
