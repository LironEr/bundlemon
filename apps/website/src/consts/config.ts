export const BUNDLEMON_SERVICE_URL = import.meta.env.VITE_BUNDLEMON_SERVICE_URL || 'http://localhost:3333';
export const GITHUB_APP_ID = import.meta.env.VITE_GITHUB_APP_ID || 'Iv1.0647108b3f7bfbd3';
