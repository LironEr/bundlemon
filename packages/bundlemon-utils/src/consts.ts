export enum Compression {
  None = 'none',
  Gzip = 'gzip',
}

export enum DiffChange {
  NoChange = 'No change',
  Update = 'Update',
  Add = 'Add',
  Remove = 'Remove',
}

export enum Status {
  Pass = 'Pass',
  Fail = 'Fail',
}

export enum FailReason {
  MaxSize = 'MaxSize',
  MaxPercentIncrease = 'MaxPercentIncrease',
}
