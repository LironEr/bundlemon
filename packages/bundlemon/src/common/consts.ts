export enum EnvVar {
  remoteFlag = 'BUNDLEMON_REMOTE',
  projectId = 'BUNDLEMON_PROJECT_ID',
  projectApiKey = 'BUNDLEMON_PROJECT_APIKEY',
  serviceURL = 'BUNDLEMON_SERVICE_URL',
  subProject = 'BUNDLEMON_SUB_PROJECT',
}

export const serviceUrl = process.env[EnvVar.serviceURL] || 'https://api.bundlemon.dev';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const packageJSON = require('../../package.json');

export const version = packageJSON.version;

export enum CreateCommitRecordAuthType {
  ProjectApiKey = 'PROJECT_API_KEY',
  GithubActions = 'GITHUB_ACTIONS',
}
