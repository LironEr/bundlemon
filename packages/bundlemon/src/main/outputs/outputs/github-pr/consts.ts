export const COMMENT_IDENTIFIER = '<!-- bundlemon -->';
