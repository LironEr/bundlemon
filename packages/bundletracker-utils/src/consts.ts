/* istanbul ignore file */

export const compressions = ['none', 'gzip'] as const;
