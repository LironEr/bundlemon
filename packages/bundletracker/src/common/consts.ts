export const serviceUrl = process.env.BUNDLETRACKER_SERVICE_URL || 'http://localhost:3333';
