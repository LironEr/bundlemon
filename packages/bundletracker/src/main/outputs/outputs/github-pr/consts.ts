export const COMMENT_IDENTIFIER = '<!-- bundletracker -->';
