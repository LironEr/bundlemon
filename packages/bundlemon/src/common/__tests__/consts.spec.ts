import * as consts from '../consts';

describe('consts', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
  });

  afterAll(() => {
    process.env = OLD_ENV;
  });

  test('default snapshot', () => {
    expect(consts).toMatchSnapshot();
  });

  test('serviceUrl env var exists', () => {
    const newUrl = 'https://my-bundlemon-service.com';
    process.env[consts.EnvVar.serviceURL] = newUrl;

    const { serviceUrl } = require('../consts');

    expect(serviceUrl).toEqual(newUrl);
  });
});
