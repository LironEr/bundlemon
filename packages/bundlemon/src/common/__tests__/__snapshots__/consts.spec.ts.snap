// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`consts default snapshot 1`] = `
Object {
  "CreateCommitRecordAuthType": Object {
    "GithubActions": "GITHUB_ACTIONS",
    "ProjectApiKey": "PROJECT_API_KEY",
  },
  "EnvVar": Object {
    "projectApiKey": "BUNDLEMON_PROJECT_APIKEY",
    "projectId": "BUNDLEMON_PROJECT_ID",
    "remoteFlag": "BUNDLEMON_REMOTE",
    "serviceURL": "BUNDLEMON_SERVICE_URL",
    "subProject": "BUNDLEMON_SUB_PROJECT",
  },
  "serviceUrl": "https://api.bundlemon.dev",
}
`;
